numba/misc/firstlinefinder.py

```diff
@@ -82,11 +83,21 @@ def get_func_body_first_lineno(pyfunc):
     co = pyfunc.__code__
     try:
         with open(co.co_filename) as fin:
-            file_content = fin.read()
+            source = fin.read()
+            offset = 0
     except (FileNotFoundError, OSError):
-        return
+        try:
+            lines, offset = inspect.getsourcelines(pyfunc)
+            source = "".join(lines)
```

Comment:
```suggestion
            source = "\n".join(lines)
```
`lines` doesn't contain the new line chars.

Reply:
The reason i didn't use `getsourcelines()` might just be my reluntance to handle indentation. If you add that logic, we might not need to read from `co.co_filename` ourselves.

Reply:
Thanks for the insight - I've added dedentation, which seems to work, but I think I'd feel more comfortable with `inspect.getsourcelines()` as the fallback for now, then once it's been used a bit in the wild to determine if there are other issues or if it's pretty stable, I'd then be happy with using `getsourcelines()` as the main / only method.